File: 2363. Merge Similar Items/Solution.cpp
#include <map>
#include <vector>
using namespace std;

class Solution {
public:
    vector<vector<int>> mergeSimilarItems(vector<vector<int>>& items1, vector<vector<int>>& items2)
    {
        map<int, int> dict;
        for (const vector<int> &v : items1)
            dict[v[0]] += v[1];
        for (const vector<int> &v : items2)
            dict[v[0]] += v[1];
        vector<vector<int>> res;
        for (const auto &[k , v] : dict)
            res.push_back({k, v});
        return res;
    }
};

File: 2363. Merge Similar Items/Solution.java
import java.util.*;

class Solution {
    public List<List<Integer>> mergeSimilarItems(int[][] items1, int[][] items2) {
        Map<Integer, Integer> map = new TreeMap<>();
        for (int[] v : items1) {
            map.put(v[0], map.getOrDefault(v[0], 0) + v[1]);
        }
        for (int[] v : items2) {
            map.put(v[0], map.getOrDefault(v[0], 0) + v[1]);
        }
        List<List<Integer>> res = new ArrayList<>();
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            res.add(Arrays.asList(entry.getKey(), entry.getValue()));
        }
        return res;
    }
}
